java/src/hu/holyoil/view/popupmenus/AsteroidPopupMenu.java

```diff
@@ -1,16 +1,113 @@
 package hu.holyoil.view.popupmenus;
 
+import hu.holyoil.crewmate.AbstractSpaceship;
+import hu.holyoil.crewmate.Settler;
 import hu.holyoil.neighbour.Asteroid;
+import hu.holyoil.neighbour.INeighbour;
 
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
 import java.awt.event.MouseEvent;
 
+/**
+ * Egy a settler sajátjától különböző asteroira történő kattintáshoz köthető popupmenu-t hozza létre.
+ */
 public class AsteroidPopupMenu extends AbstractPopupMenu {
+    PopupMenu asteroidPopupMenu;
+
     @Override
     protected void InitListeners() {
 
     }
 
-    public AsteroidPopupMenu(Asteroid asteroid, MouseEvent e) {
-        super();
+
+    /**
+     * Egy a settler sajátjától különböző asteroidra történő jobb és bal kattintást kezeli le.
+     * @param asteroid: az asteroid amire a játékos kattintott
+     * @param settler: a settler aki épp játékban van
+     * @param e
+     */
+    public AsteroidPopupMenu(Asteroid asteroid, Settler settler, MouseEvent e) {
+
+        setVisible(false);
+
+        asteroidPopupMenu = new PopupMenu();
```

Comment:
This is not needed. The popupmenu should contain the data, not a hidden popup menu inside it